tests/components/sensor/test_template.py

```diff
@@ -90,7 +86,7 @@ def test_invalid_name_does_not_create(self):
 
     def test_invalid_sensor_does_not_create(self):
         """Test invalid sensor."""
-        assert sensor.setup(self.hass, {
+        assert not bootstrap.setup_component(self.hass, 'sensor', {
```

Comment:
That's impossible. It should return `True` unless something else failed. If something else failed, then the fact that `states.all()` is empty might not be because of what you're testing. What is printed in the captured log ?